scanner/README.md

```diff
@@ -0,0 +1,69 @@
+# Scanner
+
+## Workflow (for v5, not current)
+
+In order of action:
+
+ - Scanner gets `/videos` & scan file system to list all new videos
+ - Scanner guesses as much as possible from filename/path ALONE (no external database query).
+   - Format should be:
+     ```json5
+     {
+         path: string,
+         version: number,
+         part: number | null,
+         rendering: sha(path except version & part),
+         guess: {
+             kind: movie | episode | trailer | interview | ...,
+             name: string,
+             year: number | null,
+             season?: number,
+             episode?: number,
+             ...
+          },
+     }
+     ```
+
+     - Apply remaps from lists (AnimeList + thexem). Format is now:
+     ```json5
+     {
+         path: string,
+         version: number,
+         part: number | null,
+         rendering: sha(path except version & part),
+         guess: {
+             kind: movie | episode | trailer | interview | ...,
+             name: string,
+             year: number | null,
+             season?: number,
+             episodes?: number[],
+             absolutes?: number[],
+             externalId: Record<string, {showId, season, number}[]>,
+             remap: {
+                 from: "thexem",
+                 oldSeason: number,
+                 oldEpisodes: number[],
+              },
+             ...
+          },
+     }
+     ```
+ - If kind is episode, try to find the serie's id on kyoo (using the previously fetched data from `/videos`):
+   - if another video in the list of already registered videos has the same `kind`, `name` & `year`, assume it's the same
+   - if a match is found, add to the video's json:
+   ```json5
+   {
+       entries: (uuid | slug | {
+           show: uuid | slug,
+           season: number,
+           episode: number,
+           externalId?: Record<string, {showId, season, number}> // takes priority over season/episode for matching if we have one
+       })[],
+   }
+   ```
+ - Scanner pushes everything to the api in a single post `/videos` call
+ - Api registers every video in the database
+ - For each video without an associated entry, the guess data + the video's id is sent to the Matcher via a queue.
+ - Matcher retrieves metadata from the movie/serie + ALL episodes/seasons (from an external provider)
+ - Matcher pushes every metadata to the api (if there are 1000 episodes but only 1 video, still push the 1000 episodes)
```

Comment:
I'm still undecided if this should be done
 - via an api call to post `/series` that would contain all episodes/seasons/...
 - or via a queue of the same data & having the api pull this queue to add items to the database.

v4 uses the api but requires a call per episode + per season + per series so scanning ddos the api.

v5 could benefit from it being an api since external services could want to create movies/series in kyoo since now we can display shows that do not have a video file available.

Reply:
Not sure if I understood your first point. What do you mean by `adding multiples items`?
My idea was to have a post /movies & a post /series. A single post would contain every metadata about the item (staff, studio, episodes, seasons...)

I think I'll create an API first because it's the easiest to write & test rn, and it would be cool to have it publicly available for 3rd party apps.
If the API becomes a bottleneck, we could add a queue for the matcher to use.

Reply:
I mean a single endpoint that can handle multiple kinds:

   POST /items `{ "type": "movie|serie|episode" ... other attributes... }`
   
If you decide to have a batch endpoint you can just generalize this:

   POST /items/batch
   ```
   [
     { "type": "movie" ... movie schema... },
     { "type": "serie" ... serie schema... },
     { "type": "episode" ... episode schema... }, # this will depend on serieId maybe
     { "type": "special" ... schema ... },
   ]
   ```
  
  I don't know how is the database schema, so this could be different, but I hope you get the idea